### DISCUSSION TITLE
the drag effect of different concentrations.

### DISCUSSION THREAD

**Discussion by cumtra** [nakib/elphbolt#121] [General]
Dear Nakib,
Thank you for this very useful code.
 I don't know how to set ‘dopingtype’, ‘numconc, conclist‘, ’‘numT’, and ‘Tlist’ under Namelist electrons When setting the runlevel = 0. 

Thanks！

**Comment by nakib** [comment]
Dear cumtra,

You can add something like the following in the `electrons` namelist when doing a `runlevel = 0` calculation for an n-doped system:
```
dopingtype = "n"
numconc = 5
conclist = 1e16 1e17 1e18 1e19 1e20
numT = 8
Tlist = 50.0 100.0 150.0 200.0 150.0 300.0 350.0 400.0
```

This will produce a file called _chempot-T_table.txt_ that will contain the chemical potentials corresponding to the listed concentrations for the listed temperature points.

For runlevel 1, `Zn` and/or `Zp` has to be set if you include electron-charged impurity scattering.

All of the above quantities are explained here: https://github.com/nakib/elphbolt#electrons.

Best,
Nakib

**Comment by cumtra** [comment]
Thank you very much for your answer！

Best

**Comment by cumtra** [comment]
Dear Nakib,
I want to know whether elphbolt can calculate the mean free path(MFP) of phonon. If so, how to set the parameters to calculate the MFP.

Thanks!
Best

**Reply by nakib** [reply]
Dear cumtra,

This is currently available only for the dragless full phonon BTE and through the post-processing runlevel. I have not yet implemented it for the dragful case. If you are comfortable with coding, you can hack it together pretty easily just by following what I did in the subroutine `postproc` in the file `src/bte.f90`. (Search for the term "mean-free" to jump to the relevant sections.)

For the dragless phonon BTE case, after you are done doing the phonon BTE calculation, you need to re-run `elphbolt` with the following added to the `numerics` namelist. Play around with the numbers, as necessary for your system.

```
&numerics
...

!Spectral coefficients calculation:
        runlevel = 2 !post-processing step

        ph_en_num = 200
        ph_en_min = 0.0
        ph_en_max = 0.100
        ph_mfp_npts = 500
/ 
```

The necessary files for the MFP will be generated in your output/T* directory. These are:
1. `nodrag_iterated_ph_mfps_sampling`
2. `nodrag_iterated_ph_mfps_ibz` (branch resolved)
3. `nodrag_iterated_ph_kappa_mfp_cumulative_*` (3x3 tensor for each branch [integer number in filename] and branch-summed [total in filename]) 

You can use 1 and 3 to generate a plot like
![image](https://github.com/nakib/elphbolt/assets/1834089/0efb7d90-2b9b-448c-a0be-93f241dc2bb0)

and 2 and the ibz phonon energies to generate a plot like
![image](https://github.com/nakib/elphbolt/assets/1834089/c595f27c-9cc5-45b8-99bd-2cac8c559630)

The above plots are from https://arxiv.org/pdf/2306.16980.pdf.

Best,
Nakib

**Reply by nakib** [reply]
p.s. If you are comfortable with the Julia language, you can try to write the MFP calculator in that. I plan to move all the post-processing stuff to Julia. Any community contribution will be strongly appreciated.

Best,
Nakib
